### PR TITLE
allow passing more keywords to hdf5 dataset creation

### PR DESCRIPTION
**Pull request by to266**
https://github.com/hyperspy/hyperspy/issues/978

```python
In [1]: import hyperspy.api as hs

In [2]: import numpy as np

In [3]: s = hs.signals.Spectrum(np.ones((1000,1000,100)))

In [4]: s.save('smaller.hdf5', compression_opts=9)
The smaller.hdf5 file was created

In [5]: s.save('bigger.hdf5', compression_opts=0) 
The bigger.hdf5 file was created

In [6]: !du -h {smaller,bigger}.hdf5
2.0M	smaller.hdf5
828M	bigger.hdf5

```

### CODE REVIEW

**Comment by francisco-dlp** [comment]
This requires a test.

**Comment by to266** [comment]
Test added

File: hyperspy/tests/io/test_hdf5.py [hunk 1]
```diff
@@ -211,3 +212,19 @@ def test_rgba16():
...
+
+    def setUp(self):
+        self.filename = 'testfile.hdf5'
+        BaseSignal([1, 2, 3]).save(self.filename, compression_opts=8)
+
+    def test_compression_opts(self):
+        f = h5py.File(self.filename)
+        d = f['Experiments/__unnamed__/data']
+        nt.assert_equal(d.compression_opts, 8)
+        nt.assert_equal(d.compression, 'gzip')
```
Comment (francisco-dlp) [inline hyperspy/tests/io/test_hdf5.py]: shouldn't we call `f.close` at this point?

Comment (to266) [inline hyperspy/tests/io/test_hdf5.py]: does not matter, as `h5py` closes all the files automatically upon exit. In fact I spent a substantial number of hours trying to disable it, but could not.... (not for this branch though)

Comment (francisco-dlp) [inline hyperspy/tests/io/test_hdf5.py]: Upon exiting what?

From the h5py docs: "[the file] should be closed when they are no longer in use". I think that it'll be safer to close it.

Comment (to266) [inline hyperspy/tests/io/test_hdf5.py]: What I meant is with GC. Any way, added :)

**Pull request merged**